src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -6107,7 +6107,6 @@ ProjectionNames QueryAnalyzer::resolveFunction(QueryTreeNodePtr & node, Identifi
               * Example: SELECT toTypeName(sum(number)) FROM numbers(10);
               */
             if (column && isColumnConst(*column) && !typeid_cast<const ColumnConst *>(column.get())->getDataColumn().isDummy() &&
-                !hasAggregateFunctionNodes(node) && !hasFunctionNode(node, "arrayJoin") &&
```

Comment:
There is example query in comment why it will not work without those code:
```
SELECT toTypeName(sum(number)) FROM numbers(10);
```
Here we expect to have single row in result, but if we perform constant folding we remove `sum` completely and in result we will have 10 rows.

Reply:
I think we need to move this condition to the line 6093 and we should not execute such functions at all.